src/Microsoft.DocAsCode.Build.Engine/SingleDocumentBuilder.cs

```diff
@@ -196,14 +196,32 @@ private bool GetCanIncremental(string configHash, string versionName)
                 return false;
             }
             var version = LastBuildInfo.Versions.SingleOrDefault(v => v.VersionName == versionName);
-            if (version == null || configHash != version.ConfigHash)
+            if (version == null)
             {
+                Logger.LogVerbose($"cannot build incrementally because last build didn't contain version {versionName}.");
                 return false;
             }
-
-            return CurrentBuildInfo.DocfxVersion == LastBuildInfo.DocfxVersion &&
-                CurrentBuildInfo.PluginHash == LastBuildInfo.PluginHash &&
-                CurrentBuildInfo.TemplateHash == LastBuildInfo.TemplateHash;
+            if (configHash != version.ConfigHash)
+            {
+                Logger.LogVerbose($"cannot build incrementally because config changed.");
+                return false;
+            }
+            if (CurrentBuildInfo.DocfxVersion != LastBuildInfo.DocfxVersion)
+            {
+                Logger.LogVerbose($"cannot build incrementally because docfx version changed.");
```

Comment:
it won't cost much to add the info, i'll add it

---
In reply to: [76950093](https://github.com/dotnet/docfx/pull/655#discussion_r76950093) [](ancestors = 76950093)